src/NosCore.GameObject/Providers/InventoryService/InventoryService.cs

```diff
@@ -474,31 +474,39 @@ public InventoryItemInstance RemoveItemAmountFromInventory(short amount, Guid id
             return null;
         }
 
-        public void RemoveItemAmountFromInventoryByVNum(short amount, short itemVNum)
+        public List<InventoryItemInstance> RemoveItemAmountFromInventoryByVNum(short amount, short itemVNum)
         {
+            List<InventoryItemInstance> result = new List<InventoryItemInstance>();
             var stayAmount = amount;
             foreach (var itemInstance in this.Select(s => s.Value)
                 .Where(w => w.ItemInstance.ItemVNum == itemVNum)
                 .OrderBy(o => o.ItemInstance.Amount))
             {
                 if (stayAmount == 0)
                 {
-                    return;
+                    break;
```

Comment:
Return null

Reply:
No, I need to always return my "result" var, this break is for gain performance, but now I replace it by a continue if you really don't need a break word.